### ISSUE TITLE
ResolvedRecurisveType hashCode throws NPE

### ISSUE THREAD

**Issue by voidmain** [FasterXML/jackson-databind#1305]
I'm getting a NullPointerException from the `ResolvedRecursiveType` during a `reader` call. Here's the stack trace:

```
Caused by: java.lang.NullPointerException: null
	at com.fasterxml.jackson.databind.type.ResolvedRecursiveType.equals(ResolvedRecursiveType.java:103) ~[ResolvedRecursiveType.class:2.8.0]
	at com.fasterxml.jackson.databind.type.TypeBindings$AsKey.equals(TypeBindings.java:458) ~[TypeBindings$AsKey.class:2.8.0]
	at java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1035) ~[na:1.8.0_05]
	at java.util.concurrent.ConcurrentHashMap.putIfAbsent(ConcurrentHashMap.java:1535) ~[na:1.8.0_05]
	at com.fasterxml.jackson.databind.util.LRUMap.putIfAbsent(LRUMap.java:64) ~[LRUMap.class:2.8.0]
	at com.fasterxml.jackson.databind.type.TypeFactory._fromClass(TypeFactory.java:1274) ~[TypeFactory.class:2.8.0]
	at com.fasterxml.jackson.databind.type.TypeFactory._fromParamType(TypeFactory.java:1384) ~[TypeFactory.class:2.8.0]
	at com.fasterxml.jackson.databind.type.TypeFactory._fromAny(TypeFactory.java:1154) ~[TypeFactory.class:2.8.0]
	at com.fasterxml.jackson.databind.type.TypeFactory._resolveSuperInterfaces(TypeFactory.java:1298) ~[TypeFactory.class:2.8.0]
	at com.fasterxml.jackson.databind.type.TypeFactory._fromClass(TypeFactory.java:1247) ~[TypeFactory.class:2.8.0]
	at com.fasterxml.jackson.databind.type.TypeFactory._fromAny(TypeFactory.java:1150) ~[TypeFactory.class:2.8.0]
	at com.fasterxml.jackson.databind.type.TypeFactory.constructType(TypeFactory.java:618) ~[TypeFactory.class:2.8.0]
	at com.fasterxml.jackson.databind.ObjectMapper.reader(ObjectMapper.java:3447) ~[ObjectMapper.class:2.8.0]
	at com.inversoft.mybatis.BaseJSONTypeHandler.getNullableResult(BaseJSONTypeHandler.java:40) ~[BaseJSONTypeHandler.class:0.3.0]
	at org.apache.ibatis.type.BaseTypeHandler.getResult(BaseTypeHandler.java:55) ~[BaseTypeHandler.class:3.2.7]
	at org.apache.ibatis.executor.resultset.DefaultResultSetHandler.getPropertyMappingValue(DefaultResultSetHandler.java:390) ~[DefaultResultSetHandler.class:3.2.7]
	at org.apache.ibatis.executor.resultset.DefaultResultSetHandler.applyPropertyMappings(DefaultResultSetHandler.java:364) ~[DefaultResultSetHandler.class:3.2.7]
	at org.apache.ibatis.executor.resultset.DefaultResultSetHandler.getRowValue(DefaultResultSetHandler.java:338) ~[DefaultResultSetHandler.class:3.2.7]
	at org.apache.ibatis.executor.resultset.DefaultResultSetHandler.handleRowValuesForSimpleResultMap(DefaultResultSetHandler.java:291) ~[DefaultResultSetHandler.class:3.2.7]
	at org.apache.ibatis.executor.resultset.DefaultResultSetHandler.handleRowValues(DefaultResultSetHandler.java:266) ~[DefaultResultSetHandler.class:3.2.7]
	at org.apache.ibatis.executor.resultset.DefaultResultSetHandler.handleResultSet(DefaultResultSetHandler.java:236) ~[DefaultResultSetHandler.class:3.2.7]
	at org.apache.ibatis.executor.resultset.DefaultResultSetHandler.handleResultSets(DefaultResultSetHandler.java:150) ~[DefaultResultSetHandler.class:3.2.7]
	at org.apache.ibatis.executor.statement.PreparedStatementHandler.query(PreparedStatementHandler.java:60) ~[PreparedStatementHandler.class:3.2.7]
	at org.apache.ibatis.executor.statement.RoutingStatementHandler.query(RoutingStatementHandler.java:73) ~[RoutingStatementHandler.class:3.2.7]
	at org.apache.ibatis.executor.SimpleExecutor.doQuery(SimpleExecutor.java:60) ~[SimpleExecutor.class:3.2.7]
	at org.apache.ibatis.executor.BaseExecutor.queryFromDatabase(BaseExecutor.java:267) ~[BaseExecutor.class:3.2.7]
	at org.apache.ibatis.executor.BaseExecutor.query(BaseExecutor.java:137) ~[BaseExecutor.class:3.2.7]
	at org.apache.ibatis.executor.CachingExecutor.query(CachingExecutor.java:96) ~[CachingExecutor.class:3.2.7]
	at org.apache.ibatis.executor.CachingExecutor.query(CachingExecutor.java:77) ~[CachingExecutor.class:3.2.7]
	at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:108) ~[DefaultSqlSession.class:3.2.7]
```

I'm working on JSON and a Java class to reproduce this. Once I have something, I'll update the issue.

**Comment by cowtowncoder** [comment]
Thanks -- I am pretty sure this is same as #1302, fixed for 2.8.1. And I just released 2.8.1, if you could see that it resolves your issue?

**Comment by voidmain** [comment]
It looks like it did fix it. Thanks.
